napalm/base/helpers.py

```diff
@@ -155,12 +161,19 @@ def regex_find_txt(pattern, text, default=""):
     value = re.findall(pattern, text)
     try:
         if not value:
-            raise Exception
+            logger.error("No Regex match found for pattern: %s" % (str(pattern)))
+            raise Exception("No Regex match found for pattern: %s" % (str(pattern)))
         if not isinstance(value, type(default)):
             if isinstance(value, list) and len(value) == 1:
                 value = value[0]
             value = type(default)(value)
-    except Exception:  # in case of any exception, returns default
+    except Exception as regexFindTxtErr01:  # in case of any exception, returns default
+        logger.error(
+            'errorCode="regexFindTxtErr01" in napalm.base.helpers with systemMessage="%s"\
```

Comment:
I think this is a bit superfluous, as `logger.error(str(regexFindTxtErr01), exc_info=True` should be sufficient - a good logger handler should point you to the right file and everything else you need, don't have to code that into the error message. Let's keep it simple :-)

Reply:
So it is superfluous at face value, but it also gives a lot of intrinsic information to any LEM Log and event manager that a consumer may be using to help that log and event manager automatically extract context from the log/runtime. The `<field>="value"` allows most LEMs to both recognize the data and tag it.

For example:
 -  in ELK you can use a [plugin filter](https://www.elastic.co/guide/en/logstash/current/plugins-filters-kv.html) to auto extract key words from the text 
- in Splunk you the [default key value](https://docs.splunk.com/Documentation/Splunk/8.0.2/Knowledge/Automatickey-valuefieldextractionsatsearch-time) mode is as described `key=value` 

The fields added here, error code: for reference in the library , the systemMessage: from the runtime, and the message: a human readable dictation of the raised error/event - all contribute value adds in a separate context to help enrich or tell that story to both the consumer at REPL then again at a LEM.

From a different perspective, if you are not living in a LEM I can see this being superfluous so I wanted to add some context. I'll defer to your judgement when I hear back from you as to whether this belongs here still. Thank you again for taking the time to review this 🙌👍.

Reply:
Thanks for this explanation. Yes, I understand your reasoning, though I have mixed feeling about scoping that to a specific application. At the same time, I like having as many details as possible, so I'd say let's go ahead with this! :wink: